# Deployment

- Laravel can be deployed using [Laravel Cloud](https://cloud.laravel.com/), which is the fastest way to deploy and scale production Laravel applications.
<?php if($assist->hasCloudEnabled()): ?>
- Activate the ___SINGLE_BACKTICK___deploying-to-cloud___SINGLE_BACKTICK___ skill whenever deploying to Laravel Cloud, configuring Cloud environments or resources, using the Cloud CLI, or troubleshooting Cloud deployments.
<?php endif; ?>
<?php /**PATH F:\DEV\LARAVEL\sysmanager\storage\framework\views/72afcb40935ce8917e713d097dcf4927.blade.php ENDPATH**/ ?>